pandas/tests/test_frame.py

```diff
@@ -794,6 +794,27 @@ def test_setitem_empty(self):
         result.loc[result.b.isnull(), 'a'] = result.a
         assert_frame_equal(result, df)
 
+    def test_setitem_empty_frame_with_boolean(self):
+        # Test for issue #10126
+        df = pd.DataFrame([[2, 2]], index=pd.DatetimeIndex(['2011-01-01']))
+        df2 = pd.DataFrame([[1, 1]], index=pd.DatetimeIndex(['2011-01-01']))
+
+        df[df > df2] = 47
```

Comment:
you don't need the first part of this test (e.g. before the loop), its just confusing.